### ISSUE TITLE
Confusion about metricFormat regex in Experiment Definition

### ISSUE THREAD

**Issue by cc-gb2** [kubeflow/katib#1536]
Hi, I'm trying to use the metricsFormat on StdOut to capture some metrics. They appear like this: 
```
I0519 16:28:41.434724      12 main.go:136] INFO:kubeflow_example.trainer:Epoch 3 Test set: Test-Loss: 0.216253, Accuracy: 71.59% (9332/13036)
```

I'd like to grab the Accuracy and the Test-Loss. When I try to use this `([\w|-]+)\s*:\s*((-?\d+)(\.\d+)?)` from the [documentation ](https://www.kubeflow.org/docs/components/katib/experiment/#metrics-collector) (literally just the equal sign replaced with a colon). No metrics are found. There is this example [here](https://github.com/kubeflow/katib/blob/master/examples/v1beta1/file-metricscollector-example.yaml) which uses escaped characters in the regex. This doesn't match up and it's unclear how these should be formatted. Is this because it is in a file-metrics-collector?
```
metricsCollectorSpec:
    source:
      filter:
        metricsFormat:
          - "{metricName: ([\\w|-]+), metricValue: ((-?\\d+)(\\.\\d+)?)}"
```

**Comment by johnugeorge** [comment]
1. did you have these metrics in objectiveMetricName or in additionalMetricNames? `Test-Loss` and `Accuracy`
2. currently, double backslash is required for char escape -  "([\\w|-]+)\\s*:\\s*((-?\\d+)(\\.\\d+)?)" 

with respect to the example pointed out, see how lines are printed in https://github.com/kubeflow/katib/blob/master/examples/v1beta1/pytorch-mnist/mnist.py#L66 

Related fn in code: https://github.com/kubeflow/katib/blob/master/pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector.go#L44

**Comment by cc-gb2** [comment]
Yes, I've got the Test-Loss and Accuracy stated like this in my spec. 
```
    objectiveMetricName: Accuracy 
    additionalMetricNames:
      - Train-Loss
      - Test-Loss
```
So regex would look like this correct?
```
metricsFormat
    - "([\\w|-]+)\\s*: \\s*((-?\\d+)(\\.\\d+)?)"
```

**Comment by cc-gb2** [comment]
If i check regex editor, for example here: https://regex101.com/r/ipDeja/1
It will find the groups correctly whereas the metric collector from katib won't

**Comment by cc-gb2** [comment]
I've fixed the error from #1537 by creating a new cluster. This issue with the `metricsFormat` still does not appear to work.

**Comment by andreyvelich** [comment]
@cc-gb2 Do you need first `(` and latest `)` in your regex ?
Maybe try this regex:
```
[\\w-]+:\\s+(-?\\d+)(\\.\\d+)?
```

**Comment by andreyvelich** [comment]
That's great.
FYI I was using this tool to verify it: https://regexr.com/

To check why it doesn't work, you can verify it in the Go lang with the input for the metrics collector.

We apply Regex for the log line here: https://github.com/kubeflow/katib/blob/master/pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector.go#L75.

**Comment by cc-gb2** [comment]
Hmm I went back to review this, and I was actually double logging so I was logging both these lines together for some debugging: 
INFO:kubeflow_example.trainer:Epoch 4 Test set: Test-Loss=0.132935, Accuracy=90.28% (9591/10624)
INFO:kubeflow_example.trainer:Epoch 4 Test set: Test-Loss:0.132935, Accuracy:90.28% (9591/10624)

The metrics collector was grabbing the Accuracy=90.28% and reporting it succeeded. Rerunning without the the double logging, using just a colon doesn't work.

I also tried your regex again, that also doesn't work. I had to slightly edit as the spaces have changed so changed your \s+ to be a \s*. The current regex: https://regex101.com/r/JF7eFN/1/), and in the yaml file: ` - "[\\w-]+:\\s*(-?\\d+)(\\.\\d+)?"`

This means the metricsFormatter still isn't collecting the formatting correctly. Just to confirm, the metrics collector should be embedded like this:
```
spec:
  objective:
          ....
  metricsCollectorSpec:
    source:
      filter:
        metricsFormat:
        - "[\\w-]+:\\s*(-?\\d+)(\\.\\d+)?"
```

**Comment by nbren12** [comment]
It would be great if katib provided some kind of local testing framework for the metrics collection. I'm also stuck on this and struggling with the long feedback loop of compiling a new docker image and then testing E2E.

**Comment by johnugeorge** [comment]
This is a great idea to have a framework to test docker image whether it satisfies all requirements

**Comment by nbren12** [comment]
Has this been fixed? This was the main reason I stopped using katib.
